app/models/build.rb

```diff
@@ -69,6 +69,31 @@ def file_from_repo(path, ttl: 1.hour)
     nil
   end
 
+  def import_kubernetes_roles
+    # TODO : Build -> KubernetesRoles association
+    project.roles.destroy_all
+
+    config_files = project.repository.directory_contents(git_sha, path: 'kubernetes')
+
+    config_files.map do |file|
+      file_contents = project.repository.file_contents(git_sha, file)
+      config_file = Kubernetes::RoleConfigFile.new(file_contents, file)
```

Comment:
I think it would be safer to only search for yaml files. I wouldn't be surprised if there was a readme file or some garbage in that directory.

For extra credit we could think of some validation on the yaml files (unfortunately currently it's only supported in kubectl)